### PR TITLE
Add VHA facility ids to profile

### PR DESCRIPTION
**Pull request by patrickvinograd**
Parses out the assigning facilities for any USVHA patient identifiers. The use case for this is that if assigning facilities are in a certain range, it indicates the user is actually a VA patient. 

My plan is to introduce the range-checking logic in the MhvAccount object that @saneshark is creating in his MHV account creation PR, instead of here in the User or MviProfile models, since while it is probably a universal fact of VA, it seems like more of an MHV/VHA concern. So the profile will just store the raw list of facility IDs. 

I can also merge this to the kudos-launch branch instead of master for now, but wanted to go ahead and get feedback.

### CODE REVIEW

File: lib/mvi/responses/profile_parser.rb [hunk 1]
```diff
@@ -141,24 +142,33 @@ def select_ssn_element(other_ids)
       end
 
       # MVI correlation id source id relationships:
-      # {source id}^{id type}^{assigning authority}^{assigning facility}^{id status}
+      # {source id}^{id type}^{assigning facility}^{assigning authority}^{id status}
```
Comment (patrickvinograd) [inline lib/mvi/responses/profile_parser.rb]: Per the MVI spec, this comment was transposed.

File: lib/mvi/responses/profile_parser.rb [hunk 2]
```diff
@@ -141,24 +142,33 @@ def select_ssn_element(other_ids)
...
+          vha_facility_ids: select_facilities(select_extension(ids,/^\w+\^PI\^\w+\^USVHA\^\w+$/, CORRELATION_ROOT_ID))
         }
       end
 
+      def select_ids(extensions)
+        return nil if extensions.empty?
+        extensions.map { |e| e[:extension].split('^')&.first }
+      end
+
+      def select_facilities(extensions)
```
Comment (patrickvinograd) [inline lib/mvi/responses/profile_parser.rb]: I guess these two methods could be DRYer but it seems readable enough.

**Comment by jcmeloni-usds** [comment]
@patrickvinograd just making sure i'm following -- each user will be related in some way to 0 to n facilities, and those will be their user object...it's not all facilities, right?  Just want to make sure it's not getting too bloated (I don't think it is).

**Comment by patrickvinograd** [comment]
@jcmeloni-usds Correct - the MVI profile will contain identifiers for the facilities at which a patient has received treatment, and we're adding those facilities into the profile. And my understanding is it's not even every facility down to the individual clinic level, but gets rolled up to the VAMC level.

**Comment by jcmeloni-usds** [comment]
Cooool

**Comment by kreek** [comment]
If this is going to be accessed from user, shorthand like user.icn, user.edipi, you may want to follow the delegate pattern in the user model `delegate :edipi, to: :mvi`, other than that LGTM 🚀

**Comment by patrickvinograd** [comment]
@kreek that's just convenience right, i can always do `user.va_profile.vha_facility_ids`?

**Comment by patrickvinograd** [comment]
Anticipate this only being used in one place so delegation doesn't seem critical.

**Comment by kreek** [comment]
@patrickvinograd yep, just for convenience

**Pull request merged**